packages/frontend/src/redux/slices/sign/index.js

```diff
@@ -1,4 +1,138 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import BN from 'bn.js';
+import cloneDeep from 'lodash.clonedeep';
+import { createSelector } from "reselect";
+
+import { Mixpanel } from "../../../mixpanel";
+import { wallet } from "../../../utils/wallet";
+import { showCustomAlert } from "../../actions/status";
+import { selectAccountId } from "../account";
+
 const SLICE_NAME = 'sign';
 
+export const SIGN_STATUS = {
+    IN_PROGRESS: 'in-progress',
+    NEEDS_CONFIRMATION: 'needs-confirmation',
+    RETRY_TRANSACTION: 'retry-tx',
+    SUCCESS: 'success',
+    ERROR: 'error'
+};
+
+export const RETRY_TX = {
+    INCREASE: 'increase',
+    GAS: {
+        DIFF: '25000000000000',
+        MAX: '300000000000000',
+    }
+};
+
+export const handleSignTransactions = createAsyncThunk(
+    `${SLICE_NAME}/handleSignTransactions`,
+    async (_, thunkAPI) => {
+        const { dispatch, getState } = thunkAPI;
+        let transactionsHashes;
+        const retryTxDirection = selectSignRetryTxDirection(getState());
+
+        const mixpanelName = `SIGN${retryTxDirection ? ` - RETRY - ${retryTxDirection}` : ''}`;
```

Comment:
There is only one direction; this action can now be `SIGN` or `SIGN - RETRY WITH INCREASED GAS` for clarity.

Reply:
Done.